src/modules/netinstall/NetInstallPage.cpp

```diff
@@ -0,0 +1,139 @@
+        YAML::Node groupDefinition = it->as<YAML::Node>();
+
+        QString name( tr( groupDefinition["name"].as<std::string>().c_str() ) );
+        QString description( tr( groupDefinition["description"].as<std::string>().c_str() ) );
+        QStringList packages;
+
+        for ( YAML::const_iterator it = groupDefinition["packages"].begin();
+                it != groupDefinition["packages"].end(); ++it )
+            packages.append( ( *it ).as<std::string>().c_str() );
+
+        m_groups[name].name = name;
+        m_groups[name].description = description;
+        m_groups[name].packages = packages;
+
+        if ( groupDefinition["selected"] )
+            m_groups[name].selected = groupDefinition["selected"].as<bool>();
+
+        if ( groupDefinition["hidden"] )
+            m_groups[name].hidden = groupDefinition["hidden"].as<bool>();
+
+        m_groupOrder.append( name );
+    }
+}
+
+void
+NetInstallPage::dataIsHere( KJob* job )
+{
+    if ( job->error() )
+    {
+        qDebug() << job->errorString();
+        ui->netinst_status->setText( tr( "Network Installation. (Disabled: Unable to fetch package lists, check your network connection)" ) );
+        return;
+    }
+
+    auto transferJob = dynamic_cast<KIO::StoredTransferJob*>( job );
+    Q_ASSERT( transferJob != nullptr );
+    ReadGroups( transferJob->data() );
+
+    QSignalMapper* mapper = new QSignalMapper( this );
+    foreach ( const QString& groupKey, m_groupOrder )
+    {
+        Group group = m_groups[groupKey];
+        if ( group.hidden )
+        {
+            // Do not present on view.
+            continue;
+        }
+
+        GroupSelectionWidget* groupWidget = new GroupSelectionWidget( group.name, group.description, group.packages, this );
+        m_groupWidgets.insert( groupKey, groupWidget );
+        ui->groupswidget->layout()->addWidget( groupWidget );
+
+        mapper->setMapping( groupWidget, groupKey );
+        connect( groupWidget, SIGNAL( toggled( bool ) ), mapper, SLOT( map() ) );
+    }
+
+    // TODO
+    emit checkReady( isReady() );
+}
+
+QStringList NetInstallPage::selectedPackages() const
+{
+    QStringList selectedPackages;
+
+    // Add all the packages for groups that are toggled in the view.
+    for ( auto it = m_groupWidgets.constBegin(); it != m_groupWidgets.constEnd(); it++ )
+    {
+        if ( it.value()->isToggled() )
+            selectedPackages += m_groups[it.key()].packages;
+    }
+
+    // Add all the packages for groups that are hidden but selected.
+    for ( const Group& group : m_groups.values() )
+    {
+        if ( group.hidden && group.selected )
+            selectedPackages += group.packages;
+    }
+
+    return selectedPackages;
+}
+
+void NetInstallPage::loadGroupList()
+{
+    QString confUrl(
+        Calamares::JobQueue::instance()->globalStorage()->value(
+            "groupsUrl" ).toString() );
+
+    KIO::Job* getJob = KIO::storedGet( confUrl, KIO::Reload, KIO::Overwrite | KIO::HideProgressInfo );
```

Comment:
I did not know about it. I have tried to use it, but for some reason even if setting the max. redirects allowed to 3, it does not follow the HTTP 301 redirect for Chakra's URL. There are several complaints online about this behaviour sadly :)